Airwallex/Core/Sources/Internal/AWXNetWorkManager.swift

```diff
@@ -0,0 +1,161 @@
+                            NSError.errorForAirwallexSDK(with: NSLocalizedString("Invalid URL format.", comment: "Invalid URL format."))))
+                }
+                return
+            }
+            request = URLRequest(url: url)
+            request.httpBody = try? JSONSerialization.data(withJSONObject: parameters ?? [:], options: [])
+        } else {
+            guard let url = URL(string: baseURL + urlString) else {
+                DispatchQueue.main.async {
+                    completion(
+                        .failure(
+                            NSError.errorForAirwallexSDK(with: NSLocalizedString("Invalid URL format.", comment: "Invalid URL format."))))
+                }
+                return
+            }
+            request = URLRequest(url: url)
+        }
+
+        request.httpMethod = method.rawValue
+
+        for (key, value) in headers {
+            request.setValue(value, forHTTPHeaderField: key)
+        }
+
+        if let clientSecret = AWXAPIClientConfiguration.shared().clientSecret {
+            request.setValue(clientSecret, forHTTPHeaderField: "client-secret")
+        }
+
+        session.dataTask(with: request) { data, _, error in
+            if let error {
+                DispatchQueue.main.async {
+                    completion(.failure(error))
+                }
+                return
+            }
+
+            guard let data else {
+                DispatchQueue.main.async {
+                    completion(
+                        .failure(
+                            NSError.errorForAirwallexSDK(with: NSLocalizedString("No data.", comment: "No data."))))
+                }
+                return
+            }
+
+            if let error = T.parseError(data) {
+                DispatchQueue.main.async {
+                    completion(
+                        .failure(
+                            NSError.errorForAirwallexSDK(with: Int(error.code) ?? -1, localizedDescription: error.message)))
+                }
+                return
+            }
+
+            if let decodedData = T.from(data) {
+                DispatchQueue.main.async {
+                    completion(.success(decodedData))
+                }
+            } else {
+                DispatchQueue.main.async {
+                    completion(
+                        .failure(
+                            NSError.errorForAirwallexSDK(with: NSLocalizedString("Fail to decode data.", comment: "Fail to decode data."))))
+                }
+            }
+        }.resume()
+    }
+
+    public func get<T: Codable>(
+        urlString: String,
+        parameters: [String: String]? = nil,
+        completion: @escaping (Result<T, Error>) -> Void
+    ) {
+        performRequest(
+            urlString: urlString, method: .GET, parameters: parameters, completion: completion
+        )
+    }
+
+    public func post<T: Codable>(
+        urlString: String,
+        parameters: [String: Any]? = nil,
```

Comment:
This is basically a json object for the POST request payload, shouldn't it be Encodable to instead? Because [Any] can be something can't be encoded as Json, so it's error prone. Also calling it payload might be better.